### ISSUE TITLE
inline notation of v:iterator.merge causes exception when cached

### ISSUE THREAD

**Issue by thopra** [FluidTYPO3/vhs#1356]
TYPO3: 8.7.4
VHS: 4.2.0

The v:iterator.merge ViewHelper throws an exception when used with inline notation. This only occurs when a cached template/partial is served.

Works fine:
``<v:iterator.merge  b="{char: charIndex}" useKeys="1">{employeeFormValues}</v:iterator.merge>``

Does not work:
``{v:iterator.merge(a: employeeFormValues, b: '{char: charIndex}', useKeys: 1)}``

Error:
````
Uncaught TYPO3 Exception
Unsupported input type; cannot convert to array! 

TYPO3Fluid\Fluid\Core\ViewHelper\Exception thrown in file
/vagrant/www/web/typo3conf/ext/vhs/Classes/Utility/ErrorUtility.php in line 25.

121 FluidTYPO3\Vhs\Utility\ErrorUtility::throwViewHelperException("Unsupported input type; cannot convert to array!")

/vagrant/www/web/typo3conf/ext/vhs/Classes/Traits/ArrayConsumingViewHelperTrait.php:
00101:             return $candidate;
00102:         }
00103:         ErrorUtility::throwViewHelperException('Unsupported input type; cannot convert to array!');
00104:     }
00105: 

120 FluidTYPO3\Vhs\ViewHelpers\Iterator\MergeViewHelper::arrayFromArrayOrTraversableOrCSVStatic(NULL)

/vagrant/www/web/typo3conf/ext/vhs/Classes/ViewHelpers/Iterator/MergeViewHelper.php:
00064:         RenderingContextInterface $renderingContext
00065:     ) {
00066:         $a = static::arrayFromArrayOrTraversableOrCSVStatic($renderChildrenClosure());
00067:         $b = static::arrayFromArrayOrTraversableOrCSVStatic($arguments['b'], $arguments['useKeys']);
00068:         return static::mergeArraysStatic($a, $b);

[...]
````

I guess this can be tracked down to the changes made in this commit a while ago:

https://github.com/FluidTYPO3/vhs/commit/f36078eb8fd9a2d3ae422f4eb1814136ae56e4ed#diff-457a41a3672275e988fe4d4e4a6a79ab

At least I can't see the argument "a" even being used anymore.

**Comment by thopra** [comment]
I just realized that it does work perfectly when used in the chain: 

``{employeeFormValues ->v:iterator.merge(b: \'{char: charIndex}\', useKeys: 1)}`` 

Which makes more sense than the tag based example above and makes it relatively easy to fix in existing templates.

**Comment by thopra** [comment]
Well at least in my tests, ``{v:iterator.merge(a: foo, b: bar, useKeys: 1)}`` doesn't work but ``{foo ->v:iterator.merge(b: bar, useKeys: 1)}`` does.  Only reproducible when cached. 

Is this intended?

**Comment by thopra** [comment]
I've just tried this with dev-development and the problem persists. I've also found that this only occurs when the first argument is an empty array. 
However, I wasn't able to reproduce the issue when trying to isolate it in a fresh TYPO3 installation. I'm not entirely sure, but since I can't reproduce the problem in another context it seems to be an issue with my controller, not the VHS ViewHelper itself. Never mind, thank you.
